packages/bsky/src/services/indexing/index.ts

```diff
@@ -93,6 +103,57 @@ export class IndexingService {
     }
   }
 
+  async indexRepo(did: string, commit: string) {
+    const now = new Date().toISOString()
+    const { pds, signingKey } = await this.didResolver.resolveAtpData(did)
+    const { sync } = new AtpAgent({ service: pds }).api.com.atproto
+
+    const { data: car } = await sync.getCheckout({ did, commit })
+    const { root, blocks } = await readCarWithRoot(new Uint8Array(car))
+    const storage = new MemoryBlockstore(blocks)
+    const repo = await verifyCheckoutToRepo(storage, root, did, signingKey)
```

Comment:
It kinda seems like the previously implemented `verifyCheckout` handles everything that you're doing here. Any particular reason for creating the new function & reading this as a repo instead?

Reply:
What I was missing there was the CIDs—was just avoiding recomputing them.

Reply:
oh got it got it. yup that's fine. but also maybe we could just update `verifyCheckout` to give back cids as well 👀 

those are utility methods meant for this exact case. i wrote them in isolation so wasn't quite sure what was needed so we should make them match the common use

could do an "in between" `verifyCheckoutCids` which returns a map similar to `RepoContents` but with cids at the leaves